bot/services/weather.py

```diff
@@ -0,0 +1,55 @@
+import logging
+from bot.config import config
+from aiogram import types
+import httpx
+from geopy.geocoders import Nominatim
+
+logger = logging.getLogger(__name__)
+
+OPEN_WEATHER_TOKEN = config.open_weather_token
+WEATHER_STACK_TOKEN = config.weather_stack_token
+
+
+def get_coordinates(city_name):
+    geolocator = Nominatim(user_agent="my_application")
+    location = geolocator.geocode(city_name, country_codes='me')
+    return location
+
+
+def get_temperature(json_data):
+    for key, value in json_data.items():
+        if key in ['temp', 'temperature']:
+            return value
+        elif isinstance(value, dict):
+            result = get_temperature(value)
+            if result:
+                return result
+
+
+async def get_weather(message: types.Message, city: str):
+    location = get_coordinates(city)
+    try:
+        latitude, longitude = location.latitude, location.longitude
+    except AttributeError:
+        await message.answer("Проверь название города")
+    else:
+        URLS = [('OpenWeather', 'http://api.openweathermap.org/data/2.5/weather',
+                 {'lat': latitude, 'lon': longitude, 'appid': OPEN_WEATHER_TOKEN, 'units': 'metric'}),
+                ('WeatherStack', 'http://api.weatherstack.com/current',
+                 {'access_key': WEATHER_STACK_TOKEN, 'query': city}),
+                ('OpenMeteo', 'https://api.open-meteo.com/v1/forecast',
+                 {'latitude': latitude, 'longitude': longitude, 'current_weather': 'true'})
+                ]
+
```

Comment:
coro_list = []
        for name, url, params in URLS:
            async with httpx.AsyncClient() as client:
                coro_list.append(client.get(url, params=params))
            
        responses = await asyncio.gather(**coro_list)
        data_res = [i.json() for i in responses if i.status_code==200]